packages/account-v2/src/modules/IDVForm/__tests__/idv-form.spec.tsx

```diff
@@ -29,4 +76,80 @@ describe('IDVForm', () => {
     it('should throw error when IDVform is not wrapped with Formik', () => {
         expect(() => render(<IDVForm selectedCountry={{}} />)).toThrowError();
     });
+
+    it('Should change the document type value when document type is changed', async () => {
+        renderComponent();
+        const elDocumentType = screen.getByLabelText(documentTypeLabel);
+
+        userEvent.click(elDocumentType);
+        expect(await screen.findByText('Document 1')).toBeInTheDocument();
+        userEvent.tab();
+        await waitFor(() => {
+            expect(screen.queryByText('Document 2')).not.toBeInTheDocument();
+        });
+    });
+
+    it('should render the hint messages for the selected document', async () => {
+        renderComponent();
+        const elDocumentType = screen.getByLabelText(documentTypeLabel);
+
+        jest.spyOn(utils, 'getSelectedDocumentConfigData').mockReturnValue(mockDocumentConfig);
+
+        userEvent.click(elDocumentType);
+
+        const elSelectedOption = await screen.findByText('Document 1');
+
+        userEvent.click(elSelectedOption);
+
+        expect(await screen.findByText('Example: 1234567890')).toBeInTheDocument();
+    });
+
+    it("Should hide document number field when 'I dont have any of these is chosen'", async () => {
+        const newProps = {
+            ...mockProps,
+            allowDefaultValue: true,
+        };
+
+        renderComponent(newProps);
+
+        const elDocumentTypeInput = screen.getByLabelText(documentTypeLabel);
+        const elDocumentNumberInput = screen.getByText('Enter your document number');
+
+        expect(elDocumentTypeInput).toBeVisible();
+        expect(elDocumentNumberInput).toBeVisible();
+
+        userEvent.click(elDocumentTypeInput);
+
+        const elSelectedOption = await screen.findByText("I don't have any of these");
+
+        userEvent.click(elSelectedOption);
+        await waitFor(() => {
+            expect(elDocumentNumberInput).not.toBeVisible();
+        });
+    });
+
+    it('should display option to do it later when allowIDVSkip is set', async () => {
+        const newProps = {
+            ...mockProps,
+            allowDefaultValue: true,
+            allowIDVSkip: true,
+        };
+
+        renderComponent(newProps);
+
+        const elDocumentTypeInput = screen.getByLabelText(documentTypeLabel);
+        userEvent.click(elDocumentTypeInput);
+
+        expect(await screen.findByText(defaultIDVSkipMessage));
```

Comment:
a doubt, `getByText` has preference, right?

Reply:
I'm nt sure... But I can check with the suggested function

Reply:
This works as per ur suggestion